components/external_storage/src/export.rs

```diff
@@ -13,24 +13,35 @@ use gcp::GcsStorage;
 use kvproto::brpb::{
     AzureBlobStorage, Gcs, Noop, StorageBackend, StorageBackend_oneof_backend as Backend, S3,
 };
-use tikv_util::time::{Instant, Limiter};
+use tikv_util::{
+    stream::block_on_external_io,
+    time::{Instant, Limiter},
+};
 
 use crate::{
     compression_reader_dispatcher, encrypt_wrap_reader, read_external_storage_into_file,
     record_storage_create, wrap_with_checksum_reader_if_needed, BackendConfig, ExternalData,
     ExternalStorage, HdfsStorage, LocalStorage, NoopStorage, RestoreConfig, UnpinReader,
 };
-pub fn create_storage(
+
+pub async fn create_storage_async(
     storage_backend: &StorageBackend,
     config: BackendConfig,
 ) -> io::Result<Box<dyn ExternalStorage>> {
     if let Some(backend) = &storage_backend.backend {
-        create_backend(backend, config)
+        create_backend_async(backend, config).await
     } else {
         Err(bad_storage_backend(storage_backend))
     }
 }
 
+pub fn create_storage(
```

Comment:
we might need to look at all the usages to make sure no async context is calling this blocking call. For example looks like in `router::StreamTaskHandler::new(async)` is still calling `create_storage `